validator/testcases/regex.py

```diff
@@ -558,6 +558,67 @@ def tests(self):
             compat_type='error')
 
 
+@register_generator
+class Gecko47RegexTests(CompatRegexTestHelper):
+    """Regex tests for Firefox 47 updates."""
+
+    VERSION = FX47_DEFINITION
+
+    def tests(self):
+        yield self.get_test(
+            r'\bnsIX509CertDB\b',
+            'Most methods in nsIX509CertDB had their unused arguments removed.',
+            (
+                'Most methods in nsIX509CertDB had their unused arguments '
+                'removed. See %s for more information.' % BUGZILLA_BUG % 1241646
+            ),
+            log_function=self.err.warning,
+            compat_type='error')
+
+        yield self.get_test(
+            r'\b(listTokens|listSlots)\b',
+            'listTokens() and listSlots() now return nsISimpleEnumerator instead of nsIEnumerator.',
+            (
+                'listTokens() and listSlots() now return nsISimpleEnumerator instead '
+                'of nsIEnumerator. See %s for more information.' % BUGZILLA_BUG % 1220237
+            ),
+            log_function=self.err.warning,
+            compat_type='error')
+
+        yield self.get_test(
+            r'\bgDevTools.jsm\b',
+            'The gDevTools.jsm module should no longer be used.',
+            (
+                'The gDevTools.jsm module should no longer be used. '
+                'The object should now be loaded using '
+                'require("devtools/client/framework/devtools"). See '
+                '%s for more information.' % BUGZILLA_BUG % 1188405
+            ),
+            log_function=self.err.warning,
+            compat_type='warning')
+
+        yield self.get_test(
+            r'\b(fuelIApplication|Application)\b',
```

Comment:
It does seem like it could have a lot of false positives. I added the deprecation validations in https://github.com/mozilla/amo-validator/commit/beadb38eb7ed404b4966e085e45fa5d2ac24b11c, they should probably be removed now. I was also checking `extIApplication` which according to https://github.com/mozilla/amo-validator/pull/315 was also required. I'm assuming it was removed as well.